docs/excel/excel-add-ins-shapes.md

```diff
@@ -0,0 +1,237 @@
+
+The following code sample shows an image file being loaded by a [FileReader](https://developer.mozilla.org/docs/Web/API/FileReader) as a string. The string has the metadata "base64," removed before the shape is created.
+
+```js
+// This sample creates an image as a Shape object in the worksheet.
+var myFile = document.getElementById("selectedFile");
+var reader = new FileReader();
+
+reader.onload = (event) => {
+    Excel.run(function (context) {
+        var startIndex = event.target.result.indexOf("base64,");
+        var myBase64 = event.target.result.substr(startIndex + 7);
+        var sheet = context.workbook.worksheets.getItem("MyWorksheet");
+        var image = sheet.shapes.addImage(myBase64);
+        image.name = "Image";
+        return context.sync();
+    }).catch(errorHandlerFunction);
+};
+
+// Read in the image file as a data URL.
+reader.readAsDataURL(myFile.files[0]);
+```
+
+Any `Shape` object can be converted to an image. [Shape.getAsImage](/javascript/api/excel/excel.shape#getasimage-format-) returns base64-encoded string. The image's format is specified as a [PictureFormat](/javascript/api/excel/excel.pictureformat) enum passed to `getAsImage`.
+
+### Lines
+
+A line is created with `ShapeCollection.addLine`. That method needs the left and top margins of the line's start and end points. It also takes a [ConnectorType](/javascript/api/excel/excel.connectortype) enum to specify how the line contorts between endpoints. The following code sample creates a straight line on the worksheet.
+
+```js
+// This sample creates a straight line from [200,50] to [300,150] on the worksheet
+Excel.run(function (context) {
+    var shapes = context.workbook.worksheets.getItem("MyWorksheet").shapes;
+    var line = shapes.addLine(200, 50, 300, 150, Excel.ConnectorType.straight);
+    line.name = "StraightLine";
+    return context.sync();
+}).catch(errorHandlerFunction);
+```
+
+Lines can be connected to other Shape objects. The `connectBeginShape` and `connectEndShape` methods attach the beginning and ending of a line to shapes at the specified connection points. The locations of these points vary by shape, but the `Shape.connectionSiteCount` can be used to ensure your add-in does not connect to a point that's out-of-bounds. A line is disconnected from any attached shapes using the `disconnectBeginShape` and `disconnectEndShape` methods.
+
+The following code sample connects the **"MyLine"** line to two shapes named **"LeftShape"** and **"RightShape"**.
+
+```js
+// This sample connects a line between two shapes at connection points '0' and '3'.
+Excel.run(function (context) {
+    var shapes = context.workbook.worksheets.getItem("MyWorksheet").shapes;
+    var line = shapes.getItem("MyLine").line;
+    line.connectBeginShape(shapes.getItem("LeftShape"), 0);
```

Comment:
I can't run this in Script Lab. The methods Line.connectBeginShape and Line.connectEndShape do not seem to be in the beta d.ts.

Reply:
Hmm, it's working for me (I have a sample in Script Lab that uses them). Those methods were renamed a couple weeks ago.  Please double check the libraries or use the SL sample and let me know if there's still problems.